### DISCUSSION TITLE
Approaches to mesh arrangement

### DISCUSSION THREAD

**Discussion by MatthewNielsen27** [elalish/manifold#886] [General]
Love the library, this industry was certainly lacking an open source mesh boolean library that was fast. I've looked into implementing something like this in the past. 

Have you seen this recent [paper](https://github.com/gcherchi/InteractiveAndRobustMeshBooleans)? They've formulated edge insertion in such a way that can use a modified earcut which runs in linear time. How does this relate to your approach?

---

https://www.gianmarcocherchi.com/pdf/linear_earcut.pdf
https://www.gianmarcocherchi.com/pdf/interactive_exact_booleans.pdf

**Comment by elalish** [comment]
Thanks for the papers! The earcut is strange - it appears to assume you already have a valid triangulation that you only want to modify, so it's definitely no replacement for ours which takes an arbitrary polygon-with-holes as input. It also relies on exact geometry, which would almost certainly fail with epsilon-valid (slightly overlapping) input, like we have. 

The Boolean paper is cool - would be nice to see them try it on our test set. I'm curious if it's as robust and/or any faster. I kind of doubt it though - ours is simpler and I think more parallel.
